homeassistant/components/google_photos/media_source.py

```diff
@@ -0,0 +1,291 @@
+                ),
+                media_item,
+            )
+            for media_item in media_items
+        ]
+        return source
+
+    def _async_config_entries(self) -> list[GooglePhotosConfigEntry]:
+        """Return active config entries."""
+        entries = self.hass.config_entries.async_entries(
+            DOMAIN, include_disabled=False, include_ignore=False
+        )
+        return [entry for entry in entries if entry.state == ConfigEntryState.LOADED]
+
+    def _async_config_entry(self, config_entry_id: str) -> GooglePhotosConfigEntry:
+        """Return a config entry with the specified id."""
+        entries = self._async_config_entries()
+        entry = next(
+            iter(entry for entry in entries if entry.unique_id == config_entry_id),
+            None,
+        )
+        if not entry:
+            raise BrowseError(
+                f"Could not find config entry for identifier: {config_entry_id}"
+            )
+        return entry
+
+
+def _build_account(
+    config_entry: GooglePhotosConfigEntry,
+    identifier: PhotosIdentifier,
+) -> BrowseMediaSource:
+    """Build the root node for a Google Photos account for a config entry."""
+    return BrowseMediaSource(
+        domain=DOMAIN,
+        identifier=identifier.as_string(),
+        media_class=MediaClass.DIRECTORY,
+        media_content_type=MediaClass.IMAGE,
+        title=config_entry.title,
+        can_play=False,
+        can_expand=True,
+    )
+
+
+def _build_album(title: str, identifier: PhotosIdentifier) -> BrowseMediaSource:
+    """Build an album node."""
+    return BrowseMediaSource(
+        domain=DOMAIN,
+        identifier=identifier.as_string(),
+        media_class=MediaClass.ALBUM,
+        media_content_type=MediaClass.ALBUM,
+        title=title,
+        can_play=False,
+        can_expand=True,
+    )
+
+
+def _build_media_item(
+    identifier: PhotosIdentifier, media_item: dict[str, Any]
+) -> BrowseMediaSource:
+    """Build the node for an individual photos or video."""
+    is_video = media_item["mediaMetadata"].get("video") is not None
+    return BrowseMediaSource(
+        domain=DOMAIN,
+        identifier=identifier.as_string(),
+        media_class=MediaClass.IMAGE if not is_video else MediaClass.VIDEO,
+        media_content_type=MediaType.IMAGE if not is_video else MediaType.VIDEO,
+        title=media_item["filename"],
+        can_play=is_video,
+        can_expand=False,
+        thumbnail=_media_url(media_item, THUMBNAIL_SIZE),
+    )
+
+
+def _media_url(media_item: dict[str, Any], max_size: int) -> str:
+    """Return a media item url with the specified max thumbnail size on the longest edge.
+
+    See https://developers.google.com/photos/library/guides/access-media-items#base-urls
+    """
+    width = media_item["mediaMetadata"]["width"]
+    height = media_item["mediaMetadata"]["height"]
+    key = "h" if height > width else "w"
```

Comment:
I did have a version setting h and w and couldn't convince myself what the right behavior is. I've updated to "h" since I agree the height is often what matters. For the full image, just setting it to be largeish in case someone wants a large image outside of the media player.  

I can't convince myself what the right behavior here is or that c should be used.

Reply:
I think just h is better than longest edge, at least for the full screen case. I haven't tried this integration yet but HA frontend in the grid for other images seems to already do some cropping by centering the image. That's why I think ideally for thumbnails we would specify w, h, and c that match what HA frontend does.

Reply:
I trust your judgement here from your past experience optimizing photos on displays :) I'll experiment with it.  I didn't notice the cropping until i saw some recent photos and they are cropped poorly (faces at top of the photo cut off -- didn't see if this is smart crop or not but it if is that would be useful)